### PR TITLE
Sc 45 rabbit mq server

### PR DESCRIPTION
**Pull request by DanyalYor**
Refer to the documentation and commit messages for a detailed explanation.

### CODE REVIEW

File: .github/workflows/Continuous-integration.yml [hunk 1]
```diff
@@ -40,3 +40,27 @@ jobs:
...
+        run: docker build -t unit_tests ./unit_tests/
+      
+      - name: Wait 10 seconds
+        run: sleep 10s
+
+      - name: Get logs
+        run: docker compose -f docker/docker-compose.yaml logs
+        
+      - name: Run Python Unit Tests
+        run: docker run --network rabbitnet unit_tests
```
Comment (jfho) [inline .github/workflows/Continuous-integration.yml]: Why is this a separate test? Building all the docker containers 3 times seems like a bit of a waste

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -3,12 +3,25 @@ version: '3.9'
...
+  rabbitmq:
+    image: rabbitmq:3.11-management
+    container_name: rabbitmq
+    user: ":"
+    ports:
+      - "5672:5672"
+      - "15672:15672"
+    networks:
+      - rabbitnet
+
```
Comment (jfho) [inline docker/docker-compose.yaml]: Why is this its own network?

**Review by adamjhr — DISMISSED** [review]
The third test might be an issue when the data-simulator is also publishing to rabbitmq (if it starts publishing before the test can publish). This is fine for now, i guess it will have to be changed in the data sim branch.

File: .github/workflows/Continuous-integration.yml (hunk 1 above)
Comment (DanyalYor) [reply .github/workflows/Continuous-integration.yml]: It is a separate job as it has a separate responsibility from the cypress-tests. 
Furthermore, by having it as a separate job we can parallelize the jobs thus making the overall pipeline run-time faster.
I have, however, changed the step where we build and spin up all the containers to now only do so for the rabbitmq service :)

File: docker/docker-compose.yaml (hunk 2 above)
Comment (DanyalYor) [reply docker/docker-compose.yaml]: Changed to now use the same network as the other containers

**Comment by DanyalYor** [comment]
> The third test might be an issue when the data-simulator is also publishing to rabbitmq (if it starts publishing before the test can publish). This is fine for now, i guess it will have to be changed in the data sim branch.

This is now fixed as we only build the rabbitmq image as part of the test job :)

**Review by adamjhr — APPROVED** [review]
lgtm!

**Pull request merged**